src/commands/new.ts

```diff
@@ -387,13 +388,71 @@ export default {
     // #region Run Packager Install
     // pnpm/yarn/npm install it
 
+    // check if there is a cache of node_modules using a hash of the package.json
+    function hash(str: string) {
+      return crypto.createHash("md5").update(str).digest("hex")
+    }
+
+    // check if there is a cache of node_modules using a hash of the package.json
+    const packageJsonHash = hash(packageJsonRaw)
+
+    // store hash at `~/.ignite/cache/${packageJsonHash}`
+    const cachePath = path(filesystem.homedir(), ".ignite", "cache", packageJsonHash)
```

Comment:
If it's directly in the home directory, it'll be backed up by macOS, so we'll want to put it somewhere like `~/Library/Caches/.ignite` instead. I'm not sure if there's a Node API for getting the recommended operating system cache directory?

(See also: https://github.com/yarnpkg/yarn/issues/1637)

Reply:
I would probably not worry about trying to cache this on Windows, btw.

Reply:
Yeah windows has some weird stuff with anti-virus stuff that makes moving lots of files slow, so I can add a check to only allow this on mac/linux

Reply:
That is a good catch, I'll do some research on the most light weight cache location

Reply:
Looks like [linux uses `/home/username/.cache`](https://medium.com/linuxforeveryone/clear-program-cache-files-on-ubuntu-20-10-eac6a05c29a2) as a convention

Reply:
Seems like `~/Library/Caches/` is the convention on MacOS, I'll update that

Reply:
Added a `cachePaths` constant for this: https://github.com/infinitered/ignite/pull/2080/files#diff-0092032ec4a204634552d43c12ee8d637a7d97ccb1533d5973a89646d10c0822R14-R18